packages/web-wallet/src/contexts/hooks/useTokenManagement.ts

```diff
@@ -0,0 +1,134 @@
+import { useState, useCallback } from 'react';
+import { tokenRegistryService } from '../../services/TokenRegistryService';
+import { readOnlyWalletWrapper } from '../../services/ReadOnlyWalletWrapper';
+import { fetchTokenBalance } from '../../utils/tokenLoading';
+import { TOKEN_IDS } from '@/constants';
+import { createLogger } from '../../utils/logger';
+import type { TokenInfo, TokenFilter } from '../../types/token';
+
+const log = createLogger('useTokenManagement');
+
+interface UseTokenManagementOptions {
+  isConnected: boolean;
+  network: string;
+}
+
+export function useTokenManagement(options: UseTokenManagementOptions) {
+  const { isConnected, network } = options;
+  const [registeredTokens, setRegisteredTokens] = useState<TokenInfo[]>([]);
+  const [selectedTokenFilter, setSelectedTokenFilter] = useState<TokenFilter>('tokens');
+
+  const refreshTokenBalances = useCallback(async () => {
+    if (!isConnected || !readOnlyWalletWrapper.isReady()) return;
+
+    try {
+      // Fetch balances for all tokens
+      const balanceUpdates = await Promise.all(
+        registeredTokens.map(async (token) => {
```

Comment:
Are we fetching balance for HTR in this case? Or just the other registered tokens?